### DISCUSSION TITLE
How can I get the Account Details of multiple accounts with just one click?

### DISCUSSION THREAD

**Discussion by tomalazarov** [postmanlabs/postman-flows#203] [Q&A] [answered]
Hi guys :) So I'm here with like a general question , something that I'm trying to achieve but with no success.... and I ended up here hoping you can help. So here's what I have to do but not sure how.

I have a request where I'm getting the "Accounts List". Imagine I have 5 different accounts - Savings, Checking, Credit, Loan and Deposit. 

My next request is "Account Details". In order to get the Account Details I need to pass the "accountID" of the account... which I'm taking from the first request "Accounts List" (each account from the accounts list have multiple different fields but the accountID is unique). 

 So, here's my question - **How using flows, with just like one click on Start I can get the Account Details of all 5 accounts?**

I'm trying to do something like that - after the Accounts List request I append a forEach block and iterate through all accounts > then I use the Create Data block to save in a list the accountIDs of all accounts > and after that somehow I need to pass them one by one to the Account Details request... but still didn't manage to do that (tried after the forEach block to append List - Top block and from there to take the first element (accountId) and to pass it to Account Details... and it works... but only for the first element.... then tried to somehow do a loop...connected the Account Details request back to List - Top with the hope it's gonna loop again and again until the list has no elements but this loop is not working) 

Any suggestions?

**Comment by tomalazarov** [comment]
Edit - Actually I've managed to do it somehow but got an error that I don't know how to overpass.

Here's how things work - I run the Account List request > then I iterate through all the accounts with forEach block > then in the Create data block I store all the accountIDs > then from Create Data block I directly connect it to the Account Details request and it is actually executed 5 times (that's the number of the accounts I have in my case), each time with the different accountId.

So far so good... that's basically what I was trying to do... but for each one of these Account Details request execution I got an error saying "mferrmsg: 'InvalidToken: De-serialization error'" (I have the token in a durable btw).

body:
    mferrmsg: 'InvalidToken: De-serialization error'
    requestid: 5fd89f63-eb7e-4452-916d-84cbf66230b4
    opstatus: 17005
    errmsg: Not authorized
    mfcode: Gateway-31

**Comment by tomalazarov** [comment]
Edit of the Edit... sry for the spam... but mby this somehow helps me brainstorm and think how things work... and this might be helpful to others...

In the above described case... there's no need of even Create Data... just after the For Each I'm connecting it to Account Details and setting the current accountID

**Anyways... it appears that the token that I've made in to durable is not working in this case**

**mferrmsg: 'InvalidToken: De-serialization error'**

**Comment by tomalazarov** [comment, marked as answer]
Solved :DD Don't bother with this discussion... my bad... next time I'll try a little bit harder to work things out by myself before writing here. 

Briefly...I was trying to do something stupid like assigning the value of the Token Durable that I've created to the CLAIMS_TOKEN environment variable that I have in the request itself.

**Comment by anudeepmedicharla** [comment]
@tomalazarov Glad that you were able to find a solution. 
If possible, we request you to write about our flow and how you achieved it under the 'Show and tell' category :)
